### DISCUSSION TITLE
Release two versions, one with different dependency versions

### DISCUSSION THREAD

**Discussion by marksie1988** [semantic-release/semantic-release#3174] [Q&A]
Hi all,

I have a typescript project that uses 'lit' they recently removed support for older browsers which caused some of my users on older hardware issues. 

Due to this I need a package file with the old dependencies and one with the latest dependencies (maybe a "legacy" branch that I merge changes from main except for the package file).

I then want to have the same version tag but with `-legacy` on the end (unless there is a different way to deal with this)

My project is here if that makes it easier to understand Https://GitHub.com/totaldebug/atomic-calendar-revive

I'm new to semantic release and can't get my head around the best way to achieve this.

**Comment by travi** [comment]
fellow home assistant user here, so i love seeing semantic-release in use to support your project!

unfortunately, there isnt a great answer for what you describe with semantic-release. these are our supported workflows: https://semantic-release.gitbook.io/semantic-release/usage/workflow-configuration#branch-types. the closest that i can recommend would maybe be the maintenance release approach where you maintain a branch for the legacy needs and a current/primary release branch for current versions. that would mean that the legacy needs would require changes to be backported to the maintenance release branch

**Reply by marksie1988** [reply]
Looking at maintenance release that's for adding feats or bug fixes to previous versions, that would mean if I added a breaking change to the latest version my maintenance release wouldn't get that?

**Reply by travi** [reply]
thats correct. i realize that it doesnt fully cover the case you are after, but those are the options supported by semantic-release.

to maintain parallel versions with the level of differences that you are suggesting, you're honestly looking at maintaining multiple pakcages being the most realistic approach, in my opinion. the only other alternative i can think of that is short of that would be to configure multiple instances of the npm plugin and making changes as a build step between each instance. that would put you in unsupported territory as far as the semantic-release project goes, so i cant officially recommend that sort of approach
